Node/src/TeamsChatConnector.ts

```diff
@@ -167,21 +175,29 @@ export class TeamsChatConnector extends builder.ChatConnector {
       // Filter out @mention
       var out: builder.IEvent[] = []; 
       for (var re of realEvents) {
-        let intentMessage = <IntentMessage>re;
+        let intentMessage = <builder.IMessage>re;
         // Add intent message
-        if (intentMessage.type == 'message') {
-          var atMentions = [];
-          intentMessage.intentText = intentMessage.text;
+        if (this.usingIntentText && intentMessage.type == 'message') {
+          let atMentions = [];
+          var botId: string = null;
+          if (re.address && re.address.bot && re.address.bot.id) {
+            botId = re.address.bot.id;
+          }
+
+          let strippedText = intentMessage.text;
           for(var ent of intentMessage.entities) {
-            if (ent.type == 'mention') {
+            if (ent.type === 'mention' && ent.mentioned && botId && ent.mentioned.id === botId) {
               atMentions.push(ent.text);
             }
           }
+
           if (atMentions.length != 0) {
             for(var mention of atMentions) {
-              intentMessage.intentText = intentMessage.intentText.replace(new RegExp(mention, 'g'), '');
+              strippedText = strippedText.replace(new RegExp(mention, 'g'), '');
```

Comment:
If "mention" contains special regexp characters this won't work properly. Need to escape those characters.